benchmarking/README.md

```diff
@@ -0,0 +1,1011 @@
+
+```bash
+# Create a temporary results directory
+mkdir -p /tmp/benchmark_results
+
+# Run the script directly
+python benchmarking/scripts/my_script.py \
+  --benchmark-results-path /tmp/benchmark_results \
+  --input /data/test.parquet \
+  --iterations 10
+
+# Check outputs
+ls -la /tmp/benchmark_results/
+cat /tmp/benchmark_results/params.json
+cat /tmp/benchmark_results/metrics.json
+```
+
+**Use Python debugger:**
+
+Add breakpoints to your script:
+
+```python
+import pdb; pdb.set_trace()
+```
+
+Or use `ipdb` for a better experience:
+
+```python
+import ipdb; ipdb.set_trace()
+```
+
+### Viewing Logs
+
+**Framework logs** are written to stdout/stderr and captured in the session directory.
+
+**Entry logs** (stdout/stderr from benchmark scripts) are saved to:
+
+```
+{results_dir}/{session_name}/{entry_name}/stdouterr.log
+```
+
+**View logs in real-time:**
+
+```bash
+tail -f results/my-session__2025-01-23__14-30-00/my_entry/stdouterr.log
+```
+
+### Common Issues
+
+**Issue: "Module not found" errors**
+
+- Ensure `PYTHONPATH` includes the benchmarking directory
+- In Docker, verify the Curator installation with `uv sync`
+
+**Issue: Permission denied when writing results**
+
+- Check volume mount permissions
+- Ensure the results directory is writable by the container user
+
+**Issue: Timeout errors**
+
+- Increase `timeout_s` in entry configuration
+- Check `default_timeout_s` in global configuration
+- Monitor resource usage (CPU, memory, GPU)
+
+**Issue: Out of memory**
+
+- Adjust Ray configuration (`num_cpus`, `object_spilling`)
+- Reduce dataset size or batch size
+- Monitor memory usage with `nvidia-smi` (GPU) or `top` (CPU)
+
+---
+
+## Examples
+
+### Example 1: Simple Benchmark
+
+**config.yaml:**
+
+```yaml
+results_dir: "./results"
+entries:
+  - name: quick_test
+    script: test_benchmark.py
+    args: --iterations 50
+    timeout_s: 300
+```
+
+**Run:**
+
+```bash
+python benchmarking/run.py --config config.yaml
+```
+
+### Example 2: Multiple Benchmarks with Datasets
+
+**config.yaml:**
+
+```yaml
+results_dir: "./results"
+datasets:
+  - name: sample_data
+    formats:
+      - type: parquet
+        path: /data/sample.parquet
+
+entries:
+  - name: benchmark_v1
+    script: my_benchmark.py
+    args: --input {dataset:sample_data,parquet} --algorithm v1
+
+  - name: benchmark_v2
+    script: my_benchmark.py
+    args: --input {dataset:sample_data,parquet} --algorithm v2
+```
+
+### Example 3: Production Setup with Sinks
+
+**config.yaml:**
+
+```yaml
+results_dir: "/benchmarks/results"
+default_timeout_s: 7200
+
+sinks:
+  - name: mlflow
+    tracking_uri: http://mlflow.example.com:5000
+    experiment: nightly-benchmarks
+    enabled: true
+  - name: slack
+    webhook_url: ${SLACK_WEBHOOK_URL}
+    enabled: true
+
+datasets:
+  - name: cc_large
+    formats:
+      - type: parquet
+        path: /data/common_crawl_large.parquet
+
+entries:
+  - name: cc_extraction
+    script: common_crawl_benchmark.py
```

Comment:
Is the user expected to create its own `common_crawl_benchmark.py` or should it be `test_benchmark.py` ?